packages/cli/src/cms/api/api-client.ts

```diff
@@ -254,9 +320,67 @@ export class ReuniteApiClient {
     domain: string;
     apiKey: string;
     version: string;
-    command: 'push' | 'push-status';
+    command: CommandOption;
   }) {
-    this.remotes = new RemotesApiClient(domain, apiKey, version, command);
+    this.command = command;
+    this.version = version;
+    this.apiClient = new ReuniteApiClient(this.version, this.command);
+
+    this.remotes = new RemotesApi(this.apiClient, domain, apiKey);
+  }
+
+  public reportSunsetWarnings(): void {
+    const sunsetWarnings = this.apiClient.sunsetWarnings;
+
+    if (sunsetWarnings.length) {
+      const [{ isSunsetExpired, sunsetDate }] = sunsetWarnings.sort(
+        (a: SunsetWarning, b: SunsetWarning) => {
+          if (a.isSunsetExpired && b.isSunsetExpired) {
+            if (a.sunsetDate > b.sunsetDate) {
+              // b will shutdown earlier
+              return 1;
+            } else {
+              return -1;
+            }
+          }
+
+          if (a.isSunsetExpired) {
+            // a should come before b because it is already expired
+            return -1;
+          }
+
+          if (b.isSunsetExpired) {
+            // b should come before a because it is already expired
+            return 1;
+          }
+
+          if (a.sunsetDate > b.sunsetDate) {
+            // b will shutdown earlier
+            return 1;
+          } else {
+            return -1;
+          }
+        }
+      );
```

Comment:
Looks like it could be simplified: 

```suggestion
      const [{ isSunsetExpired, sunsetDate }] = sunsetWarnings.sort(
        (a: SunsetWarning, b: SunsetWarning) => {
        // First, prioritize by expiration status
        if (a.isSunsetExpired !== b.isSunsetExpired) {
          return a.isSunsetExpired ? -1 : 1;
        }

        // If both are either expired or not, sort by sunset date
        return a.sunsetDate > b.sunsetDate ? 1 : -1;
      }
    );

```

Reply:
Much better, thanks!